internal/worker/resizable_permit_test.go

```diff
@@ -145,18 +145,139 @@ func TestPermit_Simulation(t *testing.T) {
 	}
 }
 
-func Test_Permit_AcquireChan(t *testing.T) {
-	permit := NewResizablePermit(2)
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-	permitChan, done := permit.AcquireChan(ctx)
-	select {
-	case <-permitChan:
-		assert.Equal(t, 2, permit.Quota())
+// Test_Permit_Acquire tests the basic acquire functionality
+// before each acquire will wait 100ms
+func Test_Permit_Acquire(t *testing.T) {
+
+	t.Run("acquire 1 permit", func(t *testing.T) {
+		permit := NewResizablePermit(1)
+		err := permit.Acquire(context.Background())
+		assert.NoError(t, err)
 		assert.Equal(t, 1, permit.Count())
-	case <-ctx.Done():
-		t.Error("unexpected timeout")
+	})
+
+	t.Run("acquire timeout", func(t *testing.T) {
+		permit := NewResizablePermit(1)
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
+		defer cancel()
+		time.Sleep(100 * time.Millisecond)
+		err := permit.Acquire(ctx)
+		assert.ErrorContains(t, err, "context deadline exceeded")
+		assert.Empty(t, permit.Count())
+	})
+
+	t.Run("cancel acquire", func(t *testing.T) {
+		permit := NewResizablePermit(1)
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+		err := permit.Acquire(ctx)
+		assert.ErrorContains(t, err, "canceled")
+		assert.Empty(t, permit.Count())
+	})
```

Comment:
these two tests are the same fwiw - you probably want to have one that does
- acquire (use up the whole semaphore)
- acquire again (blocks until timeout)
- make sure it didn't return immediately (elapsed time > like 5ms)